### PR TITLE
readme: introduce a migration guide.

### PR DESCRIPTION
**Pull request by warpfork**
This document discusses interfaces, APIs, and concept names from other libraries in golang, and how they map to this library.

Some references are to storage systems which are appearing in https://github.com/ipld/go-ipld-prime/pull/279 .

### CODE REVIEW

File: README_migrationGuide.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		- Be judicious about whether you actually want to do this.
+		  Plugging in the potential to experience unknown network latencies into code that's expecting predictable local lookup speeds
+		  may have undesirable performance outcomes.  (But if you want to do it, go ahead...)
+- Were you using [ipfs/go-ipld-format.DAGService](https://pkg.go.dev/github.com/ipfs/go-ipld-format#DAGService)?
+	- If you're using the `Add` and `Get` methods: those are now on `linking.LinkSystem`, as `Store` and `Load`.
+	- If you're using the `*Many` methods: those don't have direct replacements;
+	  we don't use APIs that force that style of future-aggregation anymore, because it's bad for pipelining.
+	  Just use `Store` and `Load`.
+- Were you using [ipfs/go-ipld-format.Node](https://pkg.go.dev/github.com/ipfs/go-ipld-format#Node)?
+	- That's actually a semantic doozy -- we use the word "node" _much_ differently now.
```
Comment (rvagg) [inline README_migrationGuide.md:33]: whoa boy, yeah, is it any wonder this continues to confuse people if we have multiple uses of the word that mean something entirely different but so closely?

**Review by rvagg — APPROVED** [review]
:thumbsup: good start
there's probably a slew of stuff in go-merkledag worth mentioning too, including the dag-pb stuff (already here) and even DAG "walking" which should be replaced with selectors now

File: README_migrationGuide.md (hunk 1 above)
Comment (warpfork) [reply README_migrationGuide.md:33]: Yeah, I wish... this was different, somehow.  But most of those wishes seem to involve time machines at this point.  Alas.

**Comment by warpfork** [comment]
Okay, I'm in cleanup mode today, so I'm gonna merge this.  Agree more pointers on migrating code about walking would probably be good in the future too though.

**Pull request merged**